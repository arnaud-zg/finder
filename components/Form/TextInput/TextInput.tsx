import { percent, rem } from 'csx'
import { Field } from 'formik'
import React from 'react'
import { style } from 'typestyle'
import { colors } from '../../../constants/colors'
import { ErrorField } from '../ErrorField'

const INPUT_TYPE_TEXT = 'text'

const className = style({
  borderColor: colors.gray.toHexString(),
  borderRadius: rem(0.25),
  borderStyle: 'solid',
  borderWidth: rem(0.0625),
  fontSize: rem(1),
  marginTop: rem(0.5),
  padding: rem(0.5),
  width: percent(100),
})

interface IProps extends React.HTMLProps<HTMLInputElement> {}

export const TextInput = (props: IProps) => (
  <Field name={props.name}>
    {({ field }) => (
      <>
        <input
          {...field}
          {...props}
          className={className}
          type={INPUT_TYPE_TEXT}
        />
        <ErrorField fieldName={field.name} />
      </>
    )}
  </Field>
)
