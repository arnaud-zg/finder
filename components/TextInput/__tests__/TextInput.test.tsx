import { Formik } from 'formik'
import React from 'react'
import renderer from 'react-test-renderer'
import { TextInput } from '../TextInput'

describe('TextInput', () => {
  it('renders correctly', () => {
    const testRenderer = renderer.create(
      <Formik
        initialValues={{ name: '' }}
        onSubmit={() => {}}
        render={() => <TextInput name="name" />}
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
