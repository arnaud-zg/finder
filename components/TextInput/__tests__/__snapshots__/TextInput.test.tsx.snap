// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<input
  className="f13jo2op"
  name="name"
  onBlur={[Function]}
  onChange={[Function]}
  type="text"
  value=""
/>
`;
